src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs

```diff
@@ -251,10 +241,26 @@ private static void AddAuthorityParamToRequest(AuthenticationRequestParameters a
                     throw new MsalClientException(MsalError.AuthenticationCanceledError, "WAM Account Picker did not return an account.");
                 }
 
+                bool isConsumerTenant = _webAccountProviderFactory.IsConsumerProvider(accountProvider);
                 // WAM returns the tenant here, not the full authority
-                bool isConsumerTenant = string.Equals(accountProvider.Authority, "consumers", StringComparison.OrdinalIgnoreCase);
                 wamPlugin = (isConsumerTenant && !isMsaPassthrough) ? _msaPlugin : _aadPlugin;
 
+                string transferToken = null;
+
+                if (isConsumerTenant && isMsaPassthrough)
```

Comment:
i thought 1st party was recommended to use /organizations for msa passthrough (over /common), so they won't have the consumer tenant, or what if they specify the tenant? is that possible w/msa-pt?

Reply:
Yes, the authority is /organizations or /<tenant_id>. But the logic here is: 

- get an account provider via the UI
- check if the user clicked on an MSA account
- MSA account + MSA-PT app means we go into this crazy logic